File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
export class Squares {
  constructor(count) {
    this.range = Array.from(Array(count).keys()).map(n => n + 1)
  }

  get sumOfSquares() {
    return this.range.reduce((acc, num) => acc + num ** 2);
  }

  get squareOfSum() {
    return this.range.reduce((acc, num) => acc + num) ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: solutions/zig/difference-of-squares/1/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var sum: usize = 0;
    var i: usize = 1;
    while (i <= number): (i+=1) {
        sum = sum + i;
    }
    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    var sum: usize = 0;
    for (0..number + 1) |n| {
        sum = sum + (n * n);
    }
    return sum;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
